### ISSUE TITLE
png not respecting user dims if bigger than screen (Mac)

### ISSUE THREAD

**Issue by doutriaux1** [UV-CDAT/vcs#129]
```python
import vcs
x=vcs.init(geometry={"width":3000,"height":2000})
x.open()
x.plot([1,2,3,4,3,2,1])
x.png("bad",width=3000,height=2000,units="pixels")
```
![bad](https://cloud.githubusercontent.com/assets/2781425/23246073/eb19f830-f945-11e6-9e88-2f586a28a419.png)

**Comment by aashish24** [comment]
@doutriaux1 do you want to look into this or should someone else?

**Comment by doutriaux1** [comment]
@aashish24 apparently there's nothing to do about it, short of re-running the pipeline on the bg=1 canvas when calling png. @chaosphere2112 mentioned we can't just transfer the VTK Actor to the new backend.

**Comment by aashish24** [comment]
> @chaosphere2112 mentioned we can't just transfer the VTK Actor to the new backend.

@doutriaux1 not sure if understand it correctly. 

> @aashish24 apparently there's nothing to do about it, short of re-running the pipeline on the bg=1

For OpenGL that is true (and I would think that for the rendering as well but may be not SVG). The underlying system would not let the window bigger than the window viewport.

**Comment by chaosphere2112** [comment]
@aashish24 This was referencing information that you and David Lonie told me a year or two ago about my animation solution; we can't move the entire `vtkRenderer` stack to a new render window and expect things to function correctly.

**Comment by aashish24** [comment]
> @aashish24 This was referencing information that you and David Lonie told me a year or two ago about my animation solution; we can't move the entire vtkRenderer stack to a new render window and expect things to function correctly.

thanks @chaosphere2112 that makes sense. 
actually since we are hoping to move to opengl2, this won't br an issue since if you have a vtkRenderer and you pass it to a new vtkRenderWindow it should destroy the current context objects and create new ones (typically OpenGL stuff) but i will double check on it.

**Comment by aashish24** [comment]
> @aashish24 apparently there's nothing to do about it, short of re-running the pipeline on the bg=1 canvas when calling png.

@doutriaux1 I think we can close it?

**Comment by doutriaux1** [comment]
@aashish24 let's at least add a warning to the user.

**Comment by danlipsa** [comment]
@scottwittenburg @doutriaux1 This happens with master after the context2d merge as well (on linux)

**Comment by danlipsa** [comment]
@doutriaux1 @scottwittenburg @aashish24 VTK has support for saving images with a certain magnification. That means that the window on the screen and the high resolution image should have the same ratio. The image will have higher resolution. I can add another parameter to the png function: magnification and deprecate width and height for png (issue a warning if they are used) and compute the magnification from those. What do you think?

**Comment by doutriaux1** [comment]
@danlipsa I'm not sure I follow this. @scottwittenburg has a fix at: https://github.com/CDAT/vcs/pull/376/files if @scottwittenburg thinks your idea is better then let's update @scottwittenburg branch with it.

**Comment by danlipsa** [comment]
I think it makes sense for the window and the png to have the same ratio - otherwise you could get something very different in the png

**Comment by doutriaux1** [comment]
@danlipsa, again I'm not sure exactly about what you're suggesting (link to examples?) but for the logo, I think that the idea was that the user's png would NOT be stretched in any away, but fit into the width/height box defined. We could add an option to make the log fit exactly that user-defined box and stretch the image. Is that what you're suggesting?

**Comment by danlipsa** [comment]
No. That's not for the logo, but for saving a plot as a png with higher resolution.

**Comment by danlipsa** [comment]
Take a look at the example at the beginning of the issue

**Comment by doutriaux1** [comment]
@danlipsa ok it makes more sense now... Ok according to @aashish24 comments above now that we have the new opengl "actually since we are hoping to move to opengl2, this won't br an issue since if you have a vtkRenderer and you pass it to a new vtkRenderWindow it should destroy the current context objects and create new ones (typically OpenGL stuff) but i will double check on it."

@danlipsa I trust your judgement. Feel free to pass a PR, I think what the user wants in the end is the capability to save the png (pdf, svg, etc...) at much higher res that the window shows. e.g window is opened with 800px wide but user wants to save with 3000px wide.

**Comment by doutriaux1** [comment]
after context2d when running (edited) script above, width of png is 8000 but height gets stuck at 1876.

When running with bg=1 I DO get a png that is 8000x4000 but the picture is cramped in the corner.

**Comment by doutriaux1** [comment]
![bad](https://user-images.githubusercontent.com/2781425/56383678-46797400-61cf-11e9-9270-4590a0c2e5ca.png)

**Comment by danlipsa** [comment]
I did some work on this in December: VTK did not report the new dimensions correctly after resizing. It does now. I did not get a change to look in vcs to completely fix this.
